corehq/apps/app_manager/views/modules.py

```diff
@@ -786,8 +786,13 @@ def overwrite_module_case_list(request, domain, app_id, module_unique_id):
     src_module = app.get_module_by_unique_id(module_unique_id)
     detail_type = request.POST['detail_type']
 
-    is_valid = _validate_overwrite_request(request, detail_type, dest_module_unique_ids, attrs_dict)
-    if not is_valid:
+    error_list = _validate_overwrite_request(request, detail_type, dest_module_unique_ids, attrs_dict)
+    if error_list:
+        for err in error_list:
+            messages.error(
+                request,
+                _(err)
```

Comment:
`_` doesn't do anything here. We can only translate static strings, not variables.

Reply:
oh interesting.. would it work if we used f strings like` f"{err}"` ?

Reply:
Not for translations. Looking at this more closely, the thing to do here is translate the error messages at the point they're created, inside of `_validate_overwrite_request`.

It's okay for translated strings to have variables in them, recognizing that the variables themselves won't get translated. Something like `_("My name is {name}").format(name)` is fine, that could get translated into "Me llamo {name}" and then HQ would substitute in the name, ultimately displaying "My name is Adil". But something like `_(err)` doesn't make sense because it's just a variable, there's nothing for the translator to translate.

F strings don't really work with translations. Something like `_(f"My name is {name}")` doesn't work because the substitutions are done before the string is passed to the translation function, so HQ will look for a translation for the string "My name is Adil" and won't find anything.

Reply:
Got it thanks